Add suport for SPM

tomas-bat opened the pull request:
# :pencil: Description
- This PR updates the framework and adds support for Swift Package Manager.

# :bulb: What’s new?
- Added `Package.swift`

pchmelar committed:
Add podspec file

fogatond committed:
Framework added

Commit:
M2-8649 Added optional scope init param to prevent active chat recovery

fogatond committed:
Merge pull request #1 from mluvii/feature/m2-8649-optional-scope-init-param-to-prevent-active-chat-recovery

M2-8649 Added optional scope init param to prevent active chat recovery

MartinPragPrazak committed:
added addCustomData

Commit:
changed building options to maintain compatibility

Commit:
added XCF framework and support for XCode 13

tomas-bat committed:
Add Package.swift

tommassbrand commented on MluviiChat/MluviiChat/MluviiChatLibrary.swift:
```diff
@@ -73,7 +77,7 @@ public class MluviiChat :  UIViewController, WKUIDelegate, WKNavigationDelegate,
     }
     
-    public func createWebView(url:String, companyGuid:String, tenantId:String, presetName:String?, language:String?) -> WKWebView {
+    public func createWebView(url:String, companyGuid:String, tenantId:String, presetName:String?, language:String?, scope:String?) -> WKWebView {
```
Format :) add whitespace

tommassbrand commented on MluviiChat/MluviiChat/MluviiChatLibrary.swift:
```diff
@@ -54,7 +54,7 @@ public class MluviiChat :  UIViewController, WKUIDelegate, WKNavigationDelegate,
     
-    public func createLink(url:String, companyGuid:String, tenantId:String, presetName:String? = nil, language:String? = nil) -> String {
+    public func createLink(url:String, companyGuid:String, tenantId:String, presetName:String? = nil, language:String? = nil, scope:String? = nil) -> String {
         var link: String = "https://\(url)/MobileSdkWidget?c=\(companyGuid)&t=\(tenantId)"
          var optionalPresetName = ""
          if(presetName != nil){
```
Format the lines :)

tommassbrand commented on MluviiChat/MluviiChat/MluviiChatLibrary.swift:
```diff
@@ -64,7 +64,11 @@ public class MluviiChat :  UIViewController, WKUIDelegate, WKNavigationDelegate,
         if(language != nil){
             optionalLanguage = "&l=\(language!)"
         }
-        let optionalQuery = "\(optionalPresetName)\(optionalLanguage)"
+        var optionalScope = ""
+        if(scope != nil){
+            optionalScope = "&s=\(scope!)"
+        }
```
Format the lines and conditions :)

tomas-bat commented:
@tommassbrand You commented on code from Mluvii's library, do we want to edit that?

tommassbrand commented:
Oh okay, then let it be :) @tomas-bat

pchmelar reviewed (changes requested):
- I don't know why this PR is showing changes in some files, that remain unchanged (like `MluviiChat.podspec` etc). 🤔 Probably something with how forks work on GitHub.
- `.DS_Store` and `xcuserdata` shouldn't be a part of the repo, please remove them and add them to `.gitignore` (you can copy-paste `.gitignore` from our DevStack repo) 
- Did you take a look at that SwiftUI version of this library? 🙂

tomas-bat committed:
Remove .DS_Store and xcuserdata

tomas-bat committed:
Clone gitignore from DevStack

tomas-bat committed:
Delete xcuserdata dir

tomas-bat commented:
- Removed `.DS_Store` and `xcuserdata` directory. It is still showing one of the files in this directory as edited, not deleted (not sure why, guess it's because it was also changed), but that file is not in this branch anymore.
- I did not look into the SwiftUI version very much, I just wanted to update it, add the SPM support, so we can use it as is. Should I create another task as part of the refactor, that looks into whether it can be replaced with the SwiftUI version? @pchmelar

pchmelar commented:
Yeah, please create a low-priority task for testing the SwiftUI version 🙂

tomas-bat commented:
> Yeah, please create a low-priority task for testing the SwiftUI version 🙂

Created the task [here](https://mateedevs.atlassian.net/jira/software/projects/UP/boards/1/backlog?selectedIssue=UP-650).
